app/controllers/application_controller.rb

```diff
@@ -1,10 +1,33 @@
 require './config/environment'
+require "sinatra/reloader"
 
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    register Sinatra::Reloader
+    enable :reloader
+    enable :sessions
+    set :session_secret, "secret"
   end
 
+  get '/' do
+    erb :homepage
+  end
+
+  get '/users/:user_slug' do
+    @user = User.find_by_slug(params[:user_slug])
+    erb :"users/show"
+  end
+
+  helpers do
+    def logged_in?
+      !!session[:id]
```

Comment:
It's better to rely on the successful call to `current_user` than only introspecting on the `session[:id]` value. In theory, a user might have an invalid `id` in their session which would make `logged_in?` return true even though that user would not be findable later. Also, because `current_user` is memoized, it makes sense to call it as early in the request cycle as possible, which generally is the first call `logged_in?`.

```ruby
def logged_in?
  !!current_user
end
```